models/note.go

```diff
@@ -22,7 +23,19 @@ var categoryStrings = [...]string{
 	"predictions",
 }
 
+var CannotDeserializeCategoryStringError = errors.New("String does not correspond to a Note Category")
+
+func DeserializeCategory(input string) (Category, error) {
+	for i := 0; i < len(categoryStrings); i++ {
+		if input == categoryStrings[i] {
+			return Category(i), nil
+		}
+	}
+	return MARGINALIA, CannotDeserializeCategoryStringError
```

Comment:
We should just return `0` here. I'd also be okay with returning `-1` instead of the usual `0`, because `0` is actually valid here

Reply:
actually, ignore that `-1` comment. There are always gonna be times where the falsy value is also valid. But this should just be `0` hard-coded